src/config.rs

```diff
@@ -0,0 +1,40 @@
+use crate::state::State;
+use serde::Deserialize;
+use sqlx::postgres::PgPool;
+use std::sync::Arc;
+
+#[derive(Deserialize, Clone)]
+pub struct PostgresConfig {
+    host: String,
+    user: String,
+    password: String,
+    db: String,
+}
+
+impl PostgresConfig {
+    pub fn database_url(&self) -> String {
+        format!(
+            "postgres://{}:{}@{}/{}",
+            self.user, self.password, self.host, self.db
+        )
+    }
+}
+
+#[derive(Deserialize, Clone)]
+pub struct Config {
+    pub postgres: PostgresConfig,
+}
+
+impl Config {
+    pub fn from_env() -> anyhow::Result<Self> {
+        let mut cfg = config::Config::new();
+        cfg.merge(config::Environment::new().separator("_"))?;
+        Ok(cfg.try_into()?)
+    }
```

Comment:
I think this is too complex to load a `.env` to in order to create the `Config` instance.

This could be more clean and less error prone if we use the `dotenv` crate which is already being used in this project.

`dotenv` will load environment variables straight from the file into the environment.

Then you can use `var` function from `std::env`.

Reply:
Sure!